components/SavedData.js

```diff
@@ -1,14 +1,30 @@
 import styled from 'styled-components';
+import { RiDeleteBinLine } from 'react-icons/ri';
 
 export default function SavedDataInjected({ cardData }) {
   const data = cardData;
 
+  async function removeCard() {
+    const response = await fetch('/api/savedInsulinData', {
+      method: 'DELETE',
+      headers: {
+        'content-type': 'application/json',
+      },
+
+      body: JSON.stringify(data),
+    });
+
+    const json = response.json();
+  }
   return (
     <WrapperSaved>
       <CardGrid>
         {data.map((item) => {
           return (
-            <Saved key={item.id}>
+            <Saved key={item.id} removeCard={removeCard}>
+              <IconWrapper>
+                <RiDeleteBinLine onClick={() => removeCard(item.id)} />
```

Comment:
your `Saved` component does not need the prop `removeCard`.
```suggestion
            <Saved key={item.id}>
              <IconWrapper>
                <RiDeleteBinLine onClick={() => removeCard(item.id)} />
```